keras_maskrcnn/bin/debug.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+
+"""
+Copyright 2017-2018 Fizyr (https://fizyr.com)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import os
+import sys
+import cv2
+import numpy as np
+
+# Allow relative imports when being executed as script.
+if __name__ == "__main__" and __package__ is None:
+    sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..'))
+    import keras_retinanet.bin
+    __package__ = "keras_maskrcnn.bin"
+
+# Change these to absolute imports if you copy this script outside the keras_retinanet package.
+from keras_retinanet.utils.transform import random_transform_generator
+from keras_retinanet.utils.visualization import draw_annotations, draw_boxes
+from keras_retinanet.utils.colors import label_color
+from ..utils.visualization import draw_mask
+
+
+def create_generator(args):
+    # create random transform generator for augmenting training data
+    transform_generator = random_transform_generator(
+        min_rotation=-0.1,
+        max_rotation=0.1,
+        min_translation=(-0.1, -0.1),
+        max_translation=(0.1, 0.1),
+        min_shear=-0.1,
+        max_shear=0.1,
+        min_scaling=(0.9, 0.9),
+        max_scaling=(1.1, 1.1),
+        flip_x_chance=0.5,
+        flip_y_chance=0.5,
+    )
+
+    if args.dataset_type == 'coco':
+        # import here to prevent unnecessary dependency on cocoapi
+        from ..preprocessing.coco import CocoGeneratorMask
+
+        generator = CocoGeneratorMask(
+            args.coco_path,
+            args.coco_set,
+            transform_generator=transform_generator
+        )
+    else:
+        raise ValueError('Invalid data type received: {}'.format(args.dataset_type))
+
+    return generator
+
+
+def parse_args(args):
+    parser     = argparse.ArgumentParser(description='Debug script for a Mask-RCNN network.')
```

Comment:
Maybe this should be `'Debug script for a retinanet-maskrcnn dataset.'` instead (yeah yeah I know, it's like this in retinanet too ;))